subsys/spm/Kconfig

```diff
@@ -216,6 +216,7 @@ endmenu
 if ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
 config ARM_ENTRY_VENEERS_LIB_NAME
 	string
+	prompt "Secure entry function library" if !SPM_BUILD_STRATEGY_FROM_SOURCE
```

Comment:
Note that this has no prompt because it is a duplicate of the config in zephyr/arch/arm/core/Kconfig (https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/arch/arm/core/Kconfig#L236).

This duplicate exists to change the default value to something SPM-specific.

Reply:
It seemed to work, also this is a change only needed for NCS, so to me it looks good.

Reply:
It already has a prompt? Then we shouldn't need to patch subsys/spm/Kconfig at all.

Reply:
Maybe the prompt won't be visible when ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS and NOT ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS ?

Reply:
What actually happens when there are two definitions with different prompts? Will there be two entries in menuconfig?

Reply:
No, that can't be right, could you check if you get the desired behaviour when subsys/spm/Kconfig is not patched?

Reply:
Two entries in menuconfig, yes. But if one is enabled, both are. The properties either aggregate or overwrite. Except for the depends clauses, they are OR'd together.

Reply:
Ok, so there will be a single config in the background, which can be modified from two different places in the menuconfig tree?

Reply:
I think so, but I'm not sure I've actually poked around when there are two prompts, mostly when there are two declarations of other properties.

Reply:
Two entries but you can modify one of them, and the other reflects the change.

![lotto1](https://user-images.githubusercontent.com/1547348/71784672-0fb1eb00-2ff6-11ea-8c21-ae800c3cc43c.png)

![lotto2](https://user-images.githubusercontent.com/1547348/71784675-12acdb80-2ff6-11ea-8a1b-fd22cdc87241.png)

Reply:
Thanks, that's fine with me.

Reply:
Based on reading the Kconfig sources and the below attached test it appears that we can set the veneers option without patching.

```
diff --git a/samples/nrf9160/gps/prj.conf b/samples/nrf9160/gps/prj.conf
index deb9074b..b440f990 100644
--- a/samples/nrf9160/gps/prj.conf
+++ b/samples/nrf9160/gps/prj.conf
@@ -16,3 +16,6 @@ CONFIG_MAIN_STACK_SIZE=4096
 CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_NEWLIB_LIBC=y
 CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+CONFIG_ARM_ENTRY_VENEERS_LIB_NAME="spm/yolibspmsecureentries.a"
+CONFIG_SPM=y
+CONFIG_SPM_BUILD_STRATEGY_SKIP_BUILD=y
diff --git a/subsys/spm/Kconfig b/subsys/spm/Kconfig
index 5b8e4671..28b00349 100644
--- a/subsys/spm/Kconfig
+++ b/subsys/spm/Kconfig
@@ -216,7 +216,7 @@ endmenu
 if ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
 config ARM_ENTRY_VENEERS_LIB_NAME
        string
-       prompt "Secure entry function library" if !SPM_BUILD_STRATEGY_FROM_SOURCE
+#      prompt "Secure entry function library" if !SPM_BUILD_STRATEGY_FROM_SOURCE
        default "spm/libspmsecureentries.a"
 endif
 

```

Reply:
Then we need to document to the user that this can only be set through prj.conf I guess?

Reply:
Copied prompt from original kconfig source.